plugin-bsp/src/main/kotlin/org/jetbrains/plugins/bsp/impl/magicmetamodel/impl/workspacemodel/impl/updaters/transformers/JavaModuleToDummyJavaModulesTransformerHACK.kt

```diff
@@ -80,6 +96,62 @@ public class JavaModuleToDummyJavaModulesTransformerHACK(private val projectBase
   }
 }
 
+internal fun calculateDummyResourceRootPath(
```

Comment:
**suggestion:** Consider refactoring complex function

The calculateDummyResourceRootPath function is quite complex. Consider breaking it down into smaller, more manageable functions to improve readability and maintainability.